skops/io/tests/test_persist.py

```diff
@@ -1119,3 +1120,17 @@ def test_dictionary(cls):
     loaded_obj = loads(dumps(obj))
     assert obj == loaded_obj
     assert type(obj) is cls
+
+
+def test_datetime():
+    obj = datetime.now()
+    loaded_obj = loads(dumps(obj), trusted=[datetime])
+    assert obj == loaded_obj
+    assert type(obj) is datetime
+
+
+def test_slice():
+    obj = slice(1, 2, 3)
+    loaded_obj = loads(dumps(obj))
+    assert obj == loaded_obj
+    assert type(obj) is slice
```

Comment:
Can we construct an example object that uses this `__reduce__` pattern but is not a trusted type, so that we can ensure that loading this object fails?

Reply:
the `datetime` example above fails since it's not trusted, but I can also add another test.